### PR TITLE
restore standard IO file descriptors when returning from external shell (bsc #983014)

### PR DESCRIPTION
**Pull request by wfeldt**
linuxrc was looping on closed stdin and writing to broken stdout. So, make sure IO works again when we return from sh.

### CODE REVIEW

File: util.c [hunk 1]
```diff
@@ -5457,3 +5457,40 @@ void util_set_hostname(char *hostname)
...
+
+/*
+ * Run debug shell.
+ *
+ * This takes care of screen settings and restoring standard file
+ * descriptors.
+ */
+void util_run_debugshell()
+{
+  kbd_end(1);
```
Comment (shundhammer) [inline util.c]: Really kbd_end(1) ? In both code blocks where this is copied from, it's kbd_end(0).

**Comment by shundhammer** [comment]
Other than that one remark (just trying to make sure it wasn't overlooked), LGTM.

File: util.c (hunk 1 above)
Comment (wfeldt) [inline util.c]: Yes, intentionally. The difference it that it now closes the input descriptor and reopens it later.

**Comment by shundhammer** [comment]
LGTM then.

**Pull request merged**